sourcegraph/sourcegraph
# Cody Auth: Post token to local webserver to be resilient to redirect errors

As part of the [remove authentication steps](https://docs.google.com/document/d/1AubJ226dUhaV0zC8WpA_19vzoXXtQ8AUmNqSHPaWKWQ/edit#heading=h.ftqde5tlfi9g) PRD we want to simplify the login chain a bit.

Right now, when the VS Code Cody client starts an auth processes, it will not know about the newly generated auth token until the user navigates back to VS Code. In user testing, this proofed to be confusing an some users might tab away before going through with the final redirect.

We are working on an experiment that simplifies this. The trick? When an auth process is started, start a local http server (listening on a free port on `127.0.0.1`) and forward that to the web browser. Then, when a token is created, we can `POST` the new token to that endpoint in the hope that the VS Code client.

## Test plan

https://github.com/sourcegraph/sourcegraph/assets/458591/979f311a-c3ca-4935-a6b6-55b2cbd84334

- Run a local version of Cody with [this PR](https://github.com/sourcegraph/cody/pull/3244) and `TESTING_DOTCOM_URL` set to `https://sourcegraph.test:3443`
- Have Sourcegraph running as `sg start dotcom` (make sure your `dev-private` is recent)
- In the debug version, click one of the big sign in buttons
- Notice that you do not have to wait for the redirect-back anymore
